### DISCUSSION TITLE
DNS Test upstream - Unusual behaviour

### DISCUSSION THREAD

**Discussion by adamgibbo** [AdguardTeam/AdGuardHome#3039] [Q&A]
### Issue Details

* **Version of AdGuard Home server:**
  * v0.106.0-b.5
* **How did you install AdGuard Home:**
  * Install script
* **If it's a router or IoT, please write device model:**
  * Raspberry Pi 4 Model B
* **CPU architecture:**
  * AMD64
* **Operating system and version:**
  * Debian

### Expected Behavior
When inserting DNS Server, in this case using DoH using NextDNS, then select 'Test Upstreams' AGH tests the server and gives information if successful or not.

### Actual Behavior
Even though NextDNS and ADH logs requests and DoH is working, the test shows:
'Server "https://dns.nextdns.io/*retracted*/Home": could not be used, please check that you've written it correctly'

### Screenshots

![Screenshot 2021-04-29 195143](https://user-images.githubusercontent.com/52538290/116604579-394b0000-a926-11eb-8d1e-9b58a4f0de62.jpg)

</details>

### Additional Information
Issue does not occur using regular DNS or when using smaller links only occurs with encrypted DNS (DoH DoQ and DoT).

Not working: quic://Home-*retracted*.dns.nextdns.io
Working: quic://dns.nextdns.io

Not working:  tls://Home-*retracted*.dns.nextdns.io
Working: tls://dns.nextdns.io

Not working: https://dns.nextdns.io/*retracted*/Home
Working:  https://dns.nextdns.io/

**Comment by ameshkov** [comment]
Please try using https://github.com/ameshkov/dnslookup and check what error does it show.

**Comment by adamgibbo** [comment]
Appears to be no error. I saw someone in the Telegram chat today (I think) with the same issue. Perhaps an issue with DNS testing feature?

>  `linux-arm64 → ./dnslookup google.com https://dns.nextdns.io/*retracted*/Home
dnslookup v1.4.8
dnslookup result:
;; opcode: QUERY, status: NOERROR, id: 46343
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
;; QUESTION SECTION:
;google.com.    IN       A
;; ANSWER SECTION:
google.com.     300     IN      A       142.250.200.14`

**Comment by ameshkov** [comment]
Have you tried DOT and DOQ endpoints as well?

**Comment by adamgibbo** [comment]
Yes. Weird behaviour. Added to the 'Additional information' at the bottom of OP

Queries are being answered when saving the addresses but just the "Test Upstreams" feature is broken

**Comment by adamgibbo** [comment]
If you would like a test profile I can PM you one on Telegram?

**Comment by ameshkov** [comment]
I guess that's not necessary, I can test it myself

**Comment by ameshkov** [comment]
After all, I cannot reproduce that issue.

Here's what AGH does to check the upstream - it sends a DNS query to `google-public-dns-a.google.com.` to checks that the response contains `8.8.8.8` and `4.4.4.4`. Check that your NextDNS config does not block these.

**Reply by adamgibbo** [reply]
Yes this is it. I have other DNS servers blocked on my network. Stupid me! 🤦‍♂️

![Screenshot 2021-04-29 225945](https://user-images.githubusercontent.com/52538290/116623746-9eaaeb00-a93e-11eb-86ef-5fab4eb7fb19.jpg)
